src/utils/RunSUSHI.js

```diff
@@ -11,6 +13,73 @@ const getRandomPun = utils.getRandomPun;
 const Type = utils.Type;
 const FHIRDefinitions = fhirdefs.FHIRDefinitions;
 
+async function loadAndCleanDatabase(defs, dependencies) {
+  let helperUpdate = await checkForDatabaseUpgrade(dependencies);
+  let loadExternalDependenciesReturn = { defs, emptyDependencies: [] };
+
+  if (helperUpdate.shouldUpdate) {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version + 1, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  } else {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  }
+
+  // Cleans out database of any empty objectStores
+  await cleanDatabase(loadExternalDependenciesReturn.emptyDependencies, helperUpdate.version + 2);
+  return defs;
+}
+
+/**
+ * Run GoFSH
+ * @param {array} input array of JSON definitions to be processed
+ * @param {object} options - config options for GoFSH based on user input and defaults
+ * dependencies: user set, defaults to []
+ * @param {FHIRDefinitions} testDefs - this should only be used by the unit tests so they can provide their own definitions.
+ * @returns {string} the FSH
+ */
+export async function runGoFSH(input, options, testDefs = null) {
+  // Read in the resources, either as JSON objects or as strings
+  const docs = []; // WildFHIR[]
+  input.forEach((resource, i) => {
+    const location = `Input_${i}`;
+    if (typeof resource === 'string') {
+      try {
+        resource = JSON.parse(resource);
+      } catch (e) {
+        logger.error(`Could not parse ${location} to JSON`);
+        return;
+      }
+    }
+    if (gofshUtils.isProcessableContent(resource, location)) {
+      docs.push(new processor.WildFHIR(resource, location));
+    }
+  });
+
+  // Set up the FHIRProcessor
+  const lake = new processor.LakeOfFHIR(docs);
+  let defs = testDefs || new FHIRDefinitions(); // The tests pass in their own set of definitions to use
+  const fisher = new gofshUtils.MasterFisher(lake, defs);
+  const fhirProcessor = new processor.FHIRProcessor(lake, fisher);
+
+  // Process the configuration
+  const configuration = fhirProcessor.processConfig(options.dependencies ?? []);
+
+  // Load dependencies, including those inferred from an IG file, and those given as input
+  let dependencies = configuration.config.dependencies
+    ? configuration.config.dependencies.map((dep) => `${dep.packageId}#${dep.version}`)
+    : [];
+  dependencies = sliceDependency(dependencies.join(','));
+  defs = await loadAndCleanDatabase(defs, dependencies);
+
+  // Process the FHIR to rules, and then export to FSH
+  const pkg = await gofshUtils.getResources(fhirProcessor, configuration);
+
+  // Return the string of FSH definitions
+  const fsh = new gofshExport.FSHExporter(pkg).apiExport('string'); // or use options.style
```

Comment:
I don't think we'll ever want to split into separate "files" since the FSH will only ever go into the one editor on the LHS. I think I had left it while I was coming to that conclusions, so I can take it out now! Unless you think there would be a reason to split it.

Reply:
Fixed in 487b9f1.